config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-presubmits-main.yaml

```
@@ -538,6 +538,47 @@ presubmits:
     annotations:
       testgrid-dashboards: sig-cluster-lifecycle-cluster-api-provider-azure
       testgrid-tab-name: capz-pr-e2e-upgrade-1-21-1-22
+  - name: pull-cluster-api-provider-azure-e2e-apiversion-upgrade-v1beta1
+    labels:
+      preset-dind-enabled: "true"
+      preset-kind-volume-mounts: "true"
+      preset-azure-cred-only: "true"
+      preset-azure-anonymous-pull: "true"
+    decorate: true
+    optional: true
+    always_run: false
+    branches:
+      - ^main$
+    path_alias: sigs.k8s.io/cluster-api-provider-azure
+    extra_refs:
+      - org: kubernetes
+        repo: kubernetes
+        base_ref: master
+        path_alias: k8s.io/kubernetes
+    spec:
+      containers:
+        - image: gcr.io/k8s-staging-test-infra/kubekins-e2e:v20211111-d096cb0c5f-1.22
+          args:
+            - runner.sh
+            - "./scripts/ci-e2e.sh"
+          env:
+            - name: INIT_WITH_BINARY
+              value: "https://github.com/kubernetes-sigs/cluster-api/releases/download/v0.3.23/clusterctl-{OS}-{ARCH}"
+            - name: INIT_WITH_PROVIDERS_CONTRACT
+              value: "v1alpha3"
+            - name: INIT_WITH_KUBERNETES_VERSION
+              value: "v1.21.2"
+            - name: GINKGO_FOCUS
+              value: "API Version Upgrade"
+          # we need privileged mode in order to do docker in docker
+          securityContext:
+            privileged: true
+          resources:
+            requests:
+              cpu: 7300m
+    annotations:
+      testgrid-dashboards: sig-cluster-lifecycle-cluster-api-provider-azure
+      testgrid-tab-name: capz-pr-e2e-apiversion-upgrade-main
```

```suggestion
      testgrid-tab-name: capz-pr-apiversion-upgrade-main
```

it's not actually running e2e tests on second thought